main.cpp

```diff
@@ -56,46 +56,107 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    float value;
+    const char* name;
+    T(const float v, const char* ptr) :  //1
+    value(0.0f),//2
+    name(nullptr)//3
+    {
+        value = v;
+        if(ptr != nullptr)
+        {
+            name = ptr;
+        }
+        else std::cout << "name is nullptr!\n";
+    }
 };
 
-struct <#structName1#>                                //4
-{
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+struct Comparer                                //4
+{   
+    T* compare(T* a, T* b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if (a != nullptr && b != nullptr)
+        {
+            if( a->value < b->value ) return a;
+            if( a->value > b->value ) return b;
+            std::cout << "a == b!\n";
+            return nullptr;
```

Comment:
remove this.
If either condition before this line fails, the `return nullptr` at the end of the function will be executed next
- [ ] revise